x-pack/plugins/endpoint/public/applications/endpoint/store/hosts/reducer.ts

```diff
@@ -38,30 +42,77 @@ export const hostListReducer: ImmutableReducer<HostListState, AppAction> = (
       pageSize,
       pageIndex,
       loading: false,
+      error: undefined,
+    };
+  } else if (action.type === 'serverFailedToReturnHostList') {
+    return {
+      ...state,
+      error: action.payload,
+      loading: false,
     };
   } else if (action.type === 'serverReturnedHostDetails') {
     return {
       ...state,
       details: action.payload.metadata,
+      detailsLoading: false,
+      detailsError: undefined,
     };
   } else if (action.type === 'serverFailedToReturnHostDetails') {
     return {
       ...state,
       detailsError: action.payload,
+      detailsLoading: false,
     };
-  } else if (action.type === 'userPaginatedHostList') {
-    return {
+  } else if (action.type === 'userChangedUrl') {
+    const newState: Immutable<HostState> = {
       ...state,
-      ...action.payload,
-      loading: true,
+      location: action.payload,
     };
-  } else if (action.type === 'userChangedUrl') {
+    const isCurrentlyOnListPage = isOnHostPage(newState) && !hasSelectedHost(newState);
+    const wasPreviouslyOnListPage = isOnHostPage(state) && !hasSelectedHost(state);
+    const isCurrentlyOnDetailsPage = isOnHostPage(newState) && hasSelectedHost(newState);
+    const wasPreviouslyOnDetailsPage = isOnHostPage(state) && hasSelectedHost(state);
+
+    // if on the host list page for the first time, return new location and load list
+    if (isCurrentlyOnListPage) {
+      if (!wasPreviouslyOnListPage) {
```

Comment:
we currently refresh the list if the user closes a host details flyout

Reply:
I think the logic that decides to set `loading` to `true` here should be used to decide when to actually make the request.

Reply:
as opposed to in the middleware? not really sure i understand what you mean